internal/interconnections/retrieve.go

```diff
@@ -0,0 +1,73 @@
+// Copyright © 2018 Jasmin Gacic <jasmin@stackpointcloud.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package interconnections
+
+import (
+	"context"
+	"fmt"
+
+	metal "github.com/equinix-labs/metal-go/metal/v1"
+	pager "github.com/equinix/metal-cli/internal/pagination"
+	"github.com/spf13/cobra"
+)
+
+func (c *Client) Retrieve() *cobra.Command {
+	// metrosCmd represents the metros command
+	var projectID, organizationID string
+	retrieveInterconnectionsCmd := &cobra.Command{
+		Use:     "get",
+		Aliases: []string{"list"},
+		Short:   "Retrieves a list of interconnections available to the default OrganizationID OR by project ID",
+		Long:    "Retrieves a list of interconnections available to the default Organizations and by project ID",
+		Example: `  # Lists of interconnections available to the default OrganizationID OR by project ID:	
+  metal interconnections get`,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			var interConnections []metal.Interconnection
+			var proErr error
+			inc := []string{}
+			exc := []string{}
+			if projectID != "" {
+				interConnections, proErr = pager.GetAllProjectInterconnections(c.Service, projectID, inc, exc)
```

Comment:
I think we need a new flag in this file that lets people pass in another projectID if they don't want to only check their default project.

Reply:
```
metal conn create -n "test-conn" -m da -r "primary"|"redundant" -t "dedicated"|"shared" -O <ORG_ID> -p <project_ID>
```

Reply:
That seems like it'd be right.